src/tsk_fs_attr.rs

```diff
@@ -0,0 +1,348 @@
+use std::io::{Read, Seek, SeekFrom};
+use std::convert::{TryInto, From};
+use std::ffi::CStr;
+use crate::{
+    errors::TskError,
+    tsk_fs_file::TskFsFile,
+    bindings as tsk
+};
+use std::fmt::{Display, Formatter, Result as FmtReasult};
+
+#[derive(Copy, Clone, Debug)]
+pub enum TskFsAttrType {
```

Comment:
I don't think I like having a libtsk-rs version of this when the bindings auto generate these for TSK_FS_ATTR_TYPE_ENUM.

This would require our lib to always maintain this list. Granted, I doubt it changes often, but either way, its just one more thing to upkeep and can limit people using the lib. Here you have listed 23 types, but there are actually 31 currently (thus, you can see how this is an issue):

```
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NOT_FOUND: TSK_FS_ATTR_TYPE_ENUM = 0;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_DEFAULT: TSK_FS_ATTR_TYPE_ENUM = 1;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_SI: TSK_FS_ATTR_TYPE_ENUM = 16;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_ATTRLIST: TSK_FS_ATTR_TYPE_ENUM = 32;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_FNAME: TSK_FS_ATTR_TYPE_ENUM = 48;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_VVER: TSK_FS_ATTR_TYPE_ENUM = 64;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_OBJID: TSK_FS_ATTR_TYPE_ENUM = 64;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_SEC: TSK_FS_ATTR_TYPE_ENUM = 80;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_VNAME: TSK_FS_ATTR_TYPE_ENUM = 96;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_VINFO: TSK_FS_ATTR_TYPE_ENUM = 112;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_DATA: TSK_FS_ATTR_TYPE_ENUM = 128;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_IDXROOT: TSK_FS_ATTR_TYPE_ENUM = 144;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_IDXALLOC: TSK_FS_ATTR_TYPE_ENUM = 160;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_BITMAP: TSK_FS_ATTR_TYPE_ENUM = 176;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_SYMLNK: TSK_FS_ATTR_TYPE_ENUM = 192;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_REPARSE: TSK_FS_ATTR_TYPE_ENUM = 192;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_EAINFO: TSK_FS_ATTR_TYPE_ENUM = 208;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_EA: TSK_FS_ATTR_TYPE_ENUM = 224;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_PROP: TSK_FS_ATTR_TYPE_ENUM = 240;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_NTFS_LOG: TSK_FS_ATTR_TYPE_ENUM = 256;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_UNIX_INDIR: TSK_FS_ATTR_TYPE_ENUM = 4097;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_UNIX_EXTENT: TSK_FS_ATTR_TYPE_ENUM = 4098;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_HFS_DEFAULT: TSK_FS_ATTR_TYPE_ENUM = 1;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_HFS_DATA: TSK_FS_ATTR_TYPE_ENUM = 4352;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_HFS_RSRC: TSK_FS_ATTR_TYPE_ENUM = 4353;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_HFS_EXT_ATTR: TSK_FS_ATTR_TYPE_ENUM = 4354;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_HFS_COMP_REC: TSK_FS_ATTR_TYPE_ENUM = 4355;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_APFS_DATA: TSK_FS_ATTR_TYPE_ENUM = 4352;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_APFS_RSRC: TSK_FS_ATTR_TYPE_ENUM = 4353;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_APFS_EXT_ATTR: TSK_FS_ATTR_TYPE_ENUM = 4354;
pub const TSK_FS_ATTR_TYPE_ENUM_TSK_FS_ATTR_TYPE_APFS_COMP_REC: TSK_FS_ATTR_TYPE_ENUM = 4355;
```

I think I would rather just use TSK_FS_ATTR_TYPE_ENUM instead where needed.

Reply:
I agree but I hated dealing with bunch of const. But I found a way to generate the enum as rust enum using bindgen function `rustified_enum`. I implemented then new changes